packages/oft-evm/contracts/OFTFee.sol

```diff
@@ -0,0 +1,113 @@
+     * @return amountSentLD The amount sent in local decimals.
+     * @return amountReceivedLD The amount received in local decimals on the remote.
+     */
+    function _debit(
+        address _from,
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
+        _transferFrom(_from, address(this), amountSentLD - amountReceivedLD);
+        _burn(_from, amountReceivedLD);
+    }
+
+    /**
+     * @dev Credits tokens to the specified address.
+     * @param _to The address to credit the tokens to.
+     * @param _amountLD The amount of tokens to credit in local decimals.
+     * @dev _srcEid The source chain ID.
+     * @return amountReceivedLD The amount of tokens ACTUALLY received in local decimals.
+     */
+    function _credit(
+        address _to,
+        uint256 _amountLD,
+        uint32 /*_srcEid*/
+    ) internal virtual override returns (uint256 amountReceivedLD) {
+        if (_to == address(0x0)) _to = address(0xdead); // _mint(...) does not support address(0x0)
+        // @dev Default OFT mints on dst.
+        _mint(_to, _amountLD);
+        // @dev In the case of NON-default OFT, the _amountLD MIGHT not be == amountReceivedLD.
+        return _amountLD;
+    }
+
+    /**
+     * @dev Internal function to mock the amount mutation from a OFT debit() operation.
+     * @param _amountLD The amount to send in local decimals.
+     * @param _minAmountLD The minimum amount to send in local decimals.
+     * @dev _dstEid The destination endpoint ID.
+     * @return amountSentLD The amount sent, in local decimals.
+     * @return amountReceivedLD The amount to be received on the remote chain, in local decimals.
+     *
+     * @dev This is where things like fees would be calculated and deducted from the amount to be received on the remote.
+     */
+    function _debitView(
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal view virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        uint16 feeBps = _getFeeBps(_dstEid);
+        // @dev Calculate the amount to be received on the remote chain after applying fees and removing dust.
+        amountReceivedLD = _removeDust((_amountLD * (BPS_DENOMINATOR - feeBps)) / BPS_DENOMINATOR);
+        if (amountReceivedLD < _minAmountLD) {
+            revert SlippageExceeded(amountReceivedLD, _minAmountLD);
+        }
+        amountSentLD = (amountReceivedLD * BPS_DENOMINATOR) / (BPS_DENOMINATOR - feeBps);
+    }
+
+    function _feeBalance() internal virtual override returns (uint256) {
+        return IERC20(this).balanceOf(address(this));
+    }
+
+    function _transferFrom(address _from, address _to, uint256 _amount) internal virtual override returns (uint256) {
```

Comment:
There is no transfer from, so this helper is unnecessary

Reply:
I believe it is necessary.  `Fee` is abstract and could handle anything from ERC-20 tokens to native tokens (in fact, I use native in the unit tests as it is easier to implement).  As such, how the funds are moved is implementation dependent.  We are not providing a `token` argument to `Fee` constructor, so it has no knowledge of what it is actually collecting.  It just knows that it is collecting something, and eventually someone is going to `withdrawFees()`.  It has no idea how to perform `withdrawFees()`, it just knows that the child class will provide the appropriate functionality to do so.

Additionally, `OFTFeeAdapter` can reuse `_transferFrom()` for both `debit()` and `withdrawFees()`;  the steps are identical.

I am not against simplifying this, but I would argue that the status quo is not overly complicated either.

Reply:
Yeah I understand the point, let me take another look and see if theres a way to do this differently